### ISSUE TITLE
Module init file(s) are not Python 3 compliant.

### ISSUE THREAD

**Issue by weaverba137** [desihub/desiutil#26]
The Modules distributions include init files that can be used to load functions that can be used to manipulate Modules within that language.  For example the init file for Python creates a `module()` function, such that `module('load', 'desiutil')` would be equivalent to typing `module load desiutil` on the command line.

However, the Python init file included with the two major Modules distributions that I have examined (the C version and the pure-TCL version) are not Python 3 compliant.  This breaks tests of desiutil in Python 3 because:

1. The file is meant to be loaded with `execfile()` but that is absent from Python 3.
2. The file contains the `exec` statement, which is a *function* in Python 3.  This is very similar to how `print` went from a statement to a function.

This was not noticed before, even though we run Travis tests on Python 3, because the Travis environment does not have Modules, so `MODULESHOME` is not defined, which causes the tests to be *skipped*.

Probably the easiest thing to do would be to write our own code that defines the `module()` function.  It's not a very complex function.

**Comment by weaverba137** [comment]
While fixing this I stumbled on a mind-blowing bug that I need to document.  This does not affect DESI operations, but people running older versions of Python 2.7 on their home systems may notice this.

Consider the following Python 2 code:

```python
#!/usr/bin/env python
def f():
    def g():
        exec "a = 1 + 2" in globals(), locals()
        print(a)
    return g

def h():
    def i():
        exec("a = 1 + 2", globals(), locals())
        print(a)
    return i

def main():
    a = f()
    a()
    b = h()
    b()
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main())
```
This code wouldn't be valid in Python 3, because there *only* the `exec(code, globals, locals)` form is allowed.

In Python 2, this script should print
```
3
3
```
And indeed it does, when I run it with a recent Python 2 version, like Python 2.7.11 (which is the standard python-hpcp version at NERSC).

However, with an older version, such as Python 2.7.6, I get
```
  File "test_nest", line 10
    exec("a = 1 + 2", globals(), locals())
SyntaxError: unqualified exec is not allowed in function 'i' it is a nested function
```
In this case "qualified" means that `exec` is called with the global and local variables explicitly set.

But you say, "they *are* set!"  Indeed they are, but older versions of Python have a bug that prevents them from detecting that!  This appears to have first been reported here: http://bugs.python.org/issue21591.

Bottom line: try to avoid using older versions of Python 2.7 when testing desiutil.  This problem will not exist at all when we move to Python 3.

**Comment by sbailey** [comment]
Do you know the most recent version of python that does work?  We should be explicit about our minimum python 2.7.x requirement.

I'm a bit nervous about having desiutil require modules with exec-fragilities like this.  Is this only for desiInstall?  It seems like we should be able to isolate any modules-dependent code such that desiutil is completely useable with older versions of python, and the newer versions are only required when using the modules-specific functionality.

**Comment by weaverba137** [comment]
The bug is fixed in 2.7.9, see https://hg.python.org/cpython/raw-file/v2.7.9/Misc/NEWS.

I have built, and am testing a workaround so that desiutil does not need to read the python file included with Modules at all, and the remaining exec statement is 2 & 3 compatible.

**Comment by weaverba137** [comment]
PS, the Modules functionality is *currently* only used with desiInstall, to ensure that prerequisite modules are loaded prior to compiling.  However, it is certainly conceivable that we would want to be able to manipulate Modules in Python code in the future.

**Comment by weaverba137** [comment]
Fixed by #25.
